pay-api/src/pay_api/services/direct_pay_service.py

```diff
@@ -234,19 +269,85 @@ def _get_refund_token(cls):
         return token_response
 
     @staticmethod
-    def _build_automated_refund_payload(invoice: InvoiceModel):
+    def _build_refund_revenue(paybc_invoice, refund_lines):
+        """Build PAYBC refund revenue lines for the refund."""
+        if paybc_invoice.postedrefundamount > 0 or paybc_invoice.refundedamount > 0:
+            raise BusinessException('Already refunded.')
+
+        lines = []
+        for distribution_line in refund_lines:
+            revenue_match = next((r for r in paybc_invoice.revenue
+                                  if r.revenueaccount == distribution_line.revenueaccount), None)
+            if revenue_match is None:
+                raise BusinessException('Matching distribution code to revenue account not found.')
+            if revenue_match.revenueamount < distribution_line.refund_amount:
+                raise BusinessException(f'Refund amount {1} requested exceeds revenueamount {2} in PAYBC.')
+            lines.append({'lineNumber': revenue_match.linenumber, 'refundAmount': distribution_line.refund_amount})
+        return lines
+
+    @staticmethod
+    def _build_refund_distribution_lines(refund_partial: List[RefundPartialLine]):
+        """Provide distribution lines for the refund."""
+        total = Decimal('0')
+        refund_lines = []
+        for refund in refund_partial:
+            revenue_account = None
+            if not (pli := PaymentLineItemModel.find_by_id(refund.payment_line_item_id)):
+                raise BusinessException('PLI not found.')
+            if refund.refund_amount < 0:
+                raise BusinessException('Refund amount cannot be negative.')
+            revenue_account = pli.fee_distribution_id
+            fee_distribution = DistributionCodeModel.find_by_id(pli.fee_distribution_id)
+            if refund.refund_type != RefundsPartialType.SERVICE_FEE.value:
+                if refund.refund_amount > pli.total:
+                    raise BusinessException('Refund amount { } exceeds pli total in SBC-PAY {}.')
+                revenue_account = DirectPayService._get_gl_coding(fee_distribution, refund.refund_amount)
+            elif refund.refund_type == RefundsPartialType.SERVICE_FEE.value:
+                if refund.refund_amount > pli.service_fees:
+                    raise BusinessException('Refund amount { } exceeds service_fees amount in SBC-PAY {}.')
+                service_fee_dist_id = fee_distribution.service_fee_distribution_code_id
+                service_fee_distribution = DistributionCodeModel.find_by_id(service_fee_dist_id)
+                revenue_account = DirectPayService._get_gl_coding(service_fee_distribution, refund.refund_amount)
+            refund_lines.append({
+                'revenueaccount': revenue_account,
+                'refundAmount': refund.amount
+            })
+            total += refund.amount
+        return refund_lines, total
+
+    @staticmethod
+    def _query_order_status(invoice: InvoiceModel) -> OrderStatus:
```

Comment:
Will refactor DIRECT_PAY to reuse this hopefully, not sure about the other job though... we'll see